gcc is broken on armv7l-linux

## Issue description

gcc-6.4 is broken on `master` on `armv7l-linux`. Stage 1 dies with `*** Configuration armv7l-unknown-linux-gnu not supported`.

### Steps to reproduce



## Technical details

partial log:

```
...
make[4]: Leaving directory '/tmp/nix-build-gcc-6.4.0.drv-0/build/lto-plugin'
make[3]: Leaving directory '/tmp/nix-build-gcc-6.4.0.drv-0/build/lto-plugin'
Configuring stage 1 in ./gcc
g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o expr.o -MT expr.o -MMD -MP -MF .deps/expr.Tpo ../../gcc-6.4.0/libcpp/expr.c
g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o files.o -MT files.o -MMD -MP -MF .deps/files.Tpo ../../gcc-6.4.0/libcpp/files.c
g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o identifiers.o -MT identifiers.o -MMD -MP -MF .deps/identifiers.Tpo ../../gcc-6.4.0/libcpp/identifiers.c
configure: creating cache ./config.cache
checking build system type... armv7l-unknown-linux-gnu
checking host system type... armv7l-unknown-linux-gnu
checking target system type... echo "#define LOCALEDIR \"/nix/store/rs8gxa9l046wflr9j6pdz2l9fkcfia3z-gcc-6.4.0/share/locale\"" > localedir.new
../../gcc-6.4.0/libcpp/../move-if-change localedir.new localedir.h
echo timestamp > localedir.hs
g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o lex.o -MT lex.o -MMD -MP -MF .deps/lex.Tpo ../../gcc-6.4.0/libcpp/lex.c
armv7l-unknown-linux-gnu
checking LIBRARY_PATH variable... ok
checking GCC_EXEC_PREFIX variable... ok
checking whether to place generated files in the source directory... no
checking whether a default linker was specified... no
checking whether a default assembler was specified... no
checking for armv7l-unknown-linux-gnu-gcc... gcc
checking for C compiler default output file name... g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o line-map.o -MT line-map.o -MMD -MP -MF .deps/line-map.Tpo ../../gcc-6.4.0/libcpp/line-map.c
a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables... 
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o macro.o -MT macro.o -MMD -MP -MF .deps/macro.Tpo ../../gcc-6.4.0/libcpp/macro.c
checking whether we are using the GNU C++ compiler... yes
checking whether g++ -std=gnu++98 accepts -g... yes
checking for armv7l-unknown-linux-gnu-gnatbind... no
checking for armv7l-unknown-linux-gnu-gnatmake... no
checking whether compiler driver understands Ada... no
checking how to run the C++ preprocessor... g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o mkdeps.o -MT mkdeps.o -MMD -MP -MF .deps/mkdeps.Tpo ../../gcc-6.4.0/libcpp/mkdeps.c
g++ -std=gnu++98 -E
g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o pch.o -MT pch.o -MMD -MP -MF .deps/pch.Tpo ../../gcc-6.4.0/libcpp/pch.c
checking for grep that handles long lines and -e... /nix/store/lkqf5fk1q1vysrsn7dvywr6jhkhkq46s-bootstrap-tools/bin/grep
checking for egrep... /nix/store/lkqf5fk1q1vysrsn7dvywr6jhkhkq46s-bootstrap-tools/bin/grep -E
checking for ANSI C header files... g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o symtab.o -MT symtab.o -MMD -MP -MF .deps/symtab.Tpo ../../gcc-6.4.0/libcpp/symtab.c
g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o traditional.o -MT traditional.o -MMD -MP -MF .deps/traditional.Tpo ../../gcc-6.4.0/libcpp/traditional.c
yes
checking for sys/types.h... g++ -std=gnu++98  -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include  -g -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I../../gcc-6.4.0/libcpp -I. -I../../gcc-6.4.0/libcpp/../include -I../../gcc-6.4.0/libcpp/include   -c -o init.o -MT init.o -MMD -MP -MF .deps/init.Tpo ../../gcc-6.4.0/libcpp/init.c
yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
rm -f libcpp.a
checking for strings.h... ar cru libcpp.a charset.o directives.o directives-only.o errors.o expr.o files.o identifiers.o init.o lex.o line-map.o macro.o mkdeps.o pch.o symtab.o traditional.o
ar: `u' modifier ignored since `D' is the default (see `U')
ranlib libcpp.a
make[3]: Leaving directory '/tmp/nix-build-gcc-6.4.0.drv-0/build/libcpp'
yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking how to run the C preprocessor... gcc -E
checking for inline... inline
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... 64
checking size of void *... 4
checking size of short... 2
checking size of int... 4
checking size of long... 4
checking for long long... yes
checking size of long long... 8
checking for int8_t... yes
checking for int16_t... yes
checking for int32_t... yes
checking for int64_t... yes
checking for long long int... yes
checking for intmax_t... yes
checking for intptr_t... yes
checking for uint8_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint64_t... yes
checking for unsigned long long int... yes
checking for uintmax_t... yes
checking for uintptr_t... yes
checking for int64_t underlying type... long long
checking for std::swap in <utility>... yes
checking whether g++ -std=gnu++98 is affected by placement new aliasing bug... no
checking whether g++ -std=gnu++98 supports -W... yes
checking whether g++ -std=gnu++98 supports -Wall... yes
checking whether g++ -std=gnu++98 supports -Wnarrowing... yes
checking whether g++ -std=gnu++98 supports -Wwrite-strings... yes
checking whether g++ -std=gnu++98 supports -Wcast-qual... yes
checking whether g++ -std=gnu++98 supports -Wformat... yes
checking whether gcc supports -Wstrict-prototypes... yes
checking whether gcc supports -Wmissing-prototypes... yes
checking whether g++ -std=gnu++98 supports -Wmissing-format-attribute... yes
checking whether g++ -std=gnu++98 supports -Woverloaded-virtual... yes
checking whether gcc supports -Wold-style-definition... yes
checking whether gcc supports -Wc++-compat... yes
checking whether g++ -std=gnu++98 supports -pedantic -Wlong-long -Wvariadic-macros -Woverlength-strings... yes
checking whether gcc supports -fno-exceptions... yes
checking whether gcc supports -fno-rtti... yes
checking whether gcc supports -fasynchronous-unwind-tables... yes
checking valgrind.h usability... no
checking valgrind.h presence... no
checking for valgrind.h... no
checking for VALGRIND_DISCARD in <valgrind/memcheck.h>... no
checking for VALGRIND_DISCARD in <memcheck.h>... no
checking for multiarch configuration... no, disabled auto check (configured with --native-system-header-dir)
configure: WARNING: decimal float is not supported for this target, ignored
checking whether make sets $(MAKE)... yes
checking for gawk... gawk
checking whether ln -s works... yes
checking whether ln works... yes
checking for armv7l-unknown-linux-gnu-ranlib... ranlib
checking for a BSD compatible install... /nix/store/lkqf5fk1q1vysrsn7dvywr6jhkhkq46s-bootstrap-tools/bin/install -c
checking for cmp's capabilities... gnucompare
checking for mktemp... yes
checking for makeinfo... makeinfo --split-size=5000000 --split-size=5000000
checking for modern makeinfo... yes
checking for recent Pod::Man... yes
checking for flex... /tmp/nix-build-gcc-6.4.0.drv-0/gcc-6.4.0/missing flex
checking for bison... /tmp/nix-build-gcc-6.4.0.drv-0/gcc-6.4.0/missing bison
checking for nm... nm
checking for ar... ar
checking for sphinx-build... texinfo
checking for ANSI C header files... (cached) yes
checking whether time.h and sys/time.h may both be included... yes
checking whether string.h and strings.h may both be included... yes
checking for sys/wait.h that is POSIX.1 compatible... yes
checking whether termios.h defines TIOCGWINSZ... no
checking whether sys/ioctl.h defines TIOCGWINSZ... yes
checking for limits.h... yes
checking for stddef.h... yes
checking for string.h... (cached) yes
checking for strings.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for time.h... yes
checking for iconv.h... yes
checking for fcntl.h... yes
checking for ftw.h... yes
checking for unistd.h... (cached) yes
checking for sys/file.h... yes
checking for sys/time.h... yes
checking for sys/mman.h... yes
checking for sys/resource.h... yes
checking for sys/param.h... yes
checking for sys/times.h... yes
checking for sys/stat.h... (cached) yes
checking for direct.h... no
checking for malloc.h... yes
checking for langinfo.h... yes
checking for ldfcn.h... no
checking for locale.h... yes
checking for wchar.h... yes
checking for thread.h... no
checking for pthread.h... yes
checking for CHAR_BIT... yes
checking whether byte ordering is bigendian... no
checking how to run the C++ preprocessor... g++ -std=gnu++98 -E
checking for unordered_map... no
checking for tr1/unordered_map... yes
checking for ext/hash_map... yes
checking dependency style of g++ -std=gnu++98... gcc3
checking for collect2 libraries... none required
checking for library containing exc_resume... no
checking for library containing kstat_open... no
checking for library containing ldexp... none required
checking for inttypes.h... yes
checking for times... yes
checking for clock... yes
checking for kill... yes
checking for getrlimit... yes
checking for setrlimit... yes
checking for atoq... no
checking for popen... yes
checking for sysconf... yes
checking for strsignal... yes
checking for getrusage... yes
checking for nl_langinfo... yes
checking for gettimeofday... yes
checking for mbstowcs... yes
checking for wcswidth... yes
checking for mmap... yes
checking for setlocale... yes
checking for clearerr_unlocked... yes
checking for feof_unlocked... yes
checking for ferror_unlocked... yes
checking for fflush_unlocked... yes
checking for fgetc_unlocked... yes
checking for fgets_unlocked... yes
checking for fileno_unlocked... yes
checking for fprintf_unlocked... no
checking for fputc_unlocked... yes
checking for fputs_unlocked... yes
checking for fread_unlocked... yes
checking for fwrite_unlocked... yes
checking for getchar_unlocked... yes
checking for getc_unlocked... yes
checking for putchar_unlocked... yes
checking for putc_unlocked... yes
checking for madvise... yes
checking whether mbstowcs works... yes
checking for ssize_t... yes
checking for caddr_t... yes
checking for sys/mman.h... (cached) yes
checking for mmap... (cached) yes
checking whether read-only mmap of a plain file works... yes
checking whether mmap from /dev/zero works... yes
checking for MAP_ANON(YMOUS)... yes
checking whether mmap with MAP_ANON(YMOUS) works... yes
checking for pid_t... yes
checking for vfork.h... no
checking for fork... yes
checking for vfork... yes
checking for working fork... yes
checking for working vfork... (cached) yes
checking for ld used by GCC... ld
checking if the linker (ld) is GNU ld... yes
checking for shared library run path origin... done
checking for iconv... yes
checking for iconv declaration... install-shextern size_t iconv (iconv_t cd, char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t *outbytesleft);
checking for LC_MESSAGES... yes
checking for nl_langinfo and CODESET... yes
checking whether basename is declared... yes
checking whether strstr is declared... yes
checking whether getenv is declared... yes
checking whether atol is declared... yes
checking whether atoll is declared... yes
checking whether asprintf is declared... yes
checking whether sbrk is declared... yes
checking whether abort is declared... yes
checking whether atof is declared... yes
checking whether getcwd is declared... yes
checking whether getwd is declared... yes
checking whether madvise is declared... yes
checking whether stpcpy is declared... yes
checking whether strnlen is declared... yes
checking whether strsignal is declared... yes
checking whether strverscmp is declared... yes
checking whether strtol is declared... yes
checking whether strtoul is declared... yes
checking whether strtoll is declared... yes
checking whether strtoull is declared... yes
checking whether setenv is declared... yes
checking whether unsetenv is declared... yes
checking whether errno is declared... yes
checking whether snprintf is declared... yes
checking whether vsnprintf is declared... yes
checking whether vasprintf is declared... yes
checking whether malloc is declared... yes
checking whether realloc is declared... yes
checking whether calloc is declared... yes
checking whether free is declared... yes
checking whether getopt is declared... yes
checking whether clock is declared... yes
checking whether getpagesize is declared... yes
checking whether ffs is declared... yes
checking whether clearerr_unlocked is declared... yes
checking whether feof_unlocked is declared... yes
checking whether ferror_unlocked is declared... yes
checking whether fflush_unlocked is declared... yes
checking whether fgetc_unlocked is declared... yes
checking whether fgets_unlocked is declared... yes
checking whether fileno_unlocked is declared... yes
checking whether fprintf_unlocked is declared... no
checking whether fputc_unlocked is declared... yes
checking whether fputs_unlocked is declared... yes
checking whether fread_unlocked is declared... yes
checking whether fwrite_unlocked is declared... yes
checking whether getchar_unlocked is declared... yes
checking whether getc_unlocked is declared... yes
checking whether putchar_unlocked is declared... yes
checking whether putc_unlocked is declared... yes
checking whether getrlimit is declared... yes
checking whether setrlimit is declared... yes
checking whether getrusage is declared... yes
checking whether ldgetname is declared... no
checking whether times is declared... yes
checking whether sigaltstack is declared... yes
checking for struct tms... yes
checking for clock_t... yes
checking for F_SETLKW... yes
checking if mkdir takes one argument... no
*** Configuration armv7l-unknown-linux-gnu not supported
make[2]: *** [Makefile:4181: configure-stage1-gcc] Error 1
make[2]: Leaving directory '/tmp/nix-build-gcc-6.4.0.drv-0/build'
make[1]: *** [Makefile:19436: stage1-bubble] Error 2
make[1]: Leaving directory '/tmp/nix-build-gcc-6.4.0.drv-0/build'
make: *** [Makefile:19757: bootstrap] Error 2
builder for ‘/nix/store/qkbb95wbgm0lbwrlxsd8mamxh0dn6fb5-gcc-6.4.0.drv’ failed with exit code 2
...
```

--- a/pkgs/development/compilers/gcc/4.5/default.nix
+++ b/pkgs/development/compilers/gcc/4.5/default.nix
@@ -256,7 +256,11 @@ stdenv.mkDerivation ({
     ;
 
   # TODO(@Ericson2314): Always pass "--target" and always prefix.
-  configurePlatforms = [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
+  configurePlatforms =
+    # TODO(@Ericson2314): Figure out what's going wrong with Arm
+    if hostPlatform == targetPlatform && targetPlatform.isArm
+    then []
+    else [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
 
   configureFlags =
     # Basic dependencies

--- a/pkgs/development/compilers/gcc/4.8/default.nix
+++ b/pkgs/development/compilers/gcc/4.8/default.nix
@@ -298,7 +298,11 @@ stdenv.mkDerivation ({
   dontDisableStatic = true;
 
   # TODO(@Ericson2314): Always pass "--target" and always prefix.
-  configurePlatforms = [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
+  configurePlatforms =
+    # TODO(@Ericson2314): Figure out what's going wrong with Arm
+    if hostPlatform == targetPlatform && targetPlatform.isArm
+    then []
+    else [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
 
   configureFlags =
     # Basic dependencies

--- a/pkgs/development/compilers/gcc/4.9/default.nix
+++ b/pkgs/development/compilers/gcc/4.9/default.nix
@@ -307,7 +307,11 @@ stdenv.mkDerivation ({
   dontDisableStatic = true;
 
   # TODO(@Ericson2314): Always pass "--target" and always prefix.
-  configurePlatforms = [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
+  configurePlatforms =
+    # TODO(@Ericson2314): Figure out what's going wrong with Arm
+    if hostPlatform == targetPlatform && targetPlatform.isArm
+    then []
+    else [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
 
   configureFlags =
     # Basic dependencies

--- a/pkgs/development/compilers/gcc/5/default.nix
+++ b/pkgs/development/compilers/gcc/5/default.nix
@@ -316,7 +316,11 @@ stdenv.mkDerivation ({
   dontDisableStatic = true;
 
   # TODO(@Ericson2314): Always pass "--target" and always prefix.
-  configurePlatforms = [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
+  configurePlatforms =
+    # TODO(@Ericson2314): Figure out what's going wrong with Arm
+    if hostPlatform == targetPlatform && targetPlatform.isArm
+    then []
+    else [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
 
   configureFlags =
     # Basic dependencies

--- a/pkgs/development/compilers/gcc/6/default.nix
+++ b/pkgs/development/compilers/gcc/6/default.nix
@@ -319,7 +319,11 @@ stdenv.mkDerivation ({
   dontDisableStatic = true;
 
   # TODO(@Ericson2314): Always pass "--target" and always prefix.
-  configurePlatforms = [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
+  configurePlatforms =
+    # TODO(@Ericson2314): Figure out what's going wrong with Arm
+    if hostPlatform == targetPlatform && targetPlatform.isArm
+    then []
+    else [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
 
   configureFlags =
     # Basic dependencies

--- a/pkgs/development/compilers/gcc/7/default.nix
+++ b/pkgs/development/compilers/gcc/7/default.nix
@@ -312,7 +312,11 @@ stdenv.mkDerivation ({
   dontDisableStatic = true;
 
   # TODO(@Ericson2314): Always pass "--target" and always prefix.
-  configurePlatforms = [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
+  configurePlatforms =
+    # TODO(@Ericson2314): Figure out what's going wrong with Arm
+    if hostPlatform == targetPlatform && targetPlatform.isArm
+    then []
+    else [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
 
   configureFlags =
     # Basic dependencies

--- a/pkgs/development/compilers/gcc/snapshot/default.nix
+++ b/pkgs/development/compilers/gcc/snapshot/default.nix
@@ -299,7 +299,11 @@ stdenv.mkDerivation ({
   dontDisableStatic = true;
 
   # TODO(@Ericson2314): Always pass "--target" and always prefix.
-  configurePlatforms = [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
+  configurePlatforms =
+    # TODO(@Ericson2314): Figure out what's going wrong with Arm
+    if hostPlatform == targetPlatform && targetPlatform.isArm
+    then []
+    else [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
 
   configureFlags =
     # Basic dependencies

--- a/pkgs/development/tools/misc/binutils/default.nix
+++ b/pkgs/development/tools/misc/binutils/default.nix
@@ -91,7 +91,11 @@ stdenv.mkDerivation rec {
     else "-static-libgcc";
 
   # TODO(@Ericson2314): Always pass "--target" and always targetPrefix.
-  configurePlatforms = [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
+  configurePlatforms =
+    # TODO(@Ericson2314): Figure out what's going wrong with Arm
+    if hostPlatform == targetPlatform && targetPlatform.isArm
+    then []
+    else [ "build" "host" ] ++ stdenv.lib.optional (targetPlatform != hostPlatform) "target";
 
   configureFlags = [
     "--enable-targets=all" "--enable-64-bit-bfd"

